README.rst

```diff
@@ -32,10 +32,108 @@ Issue for its creation at https://github.com/unfoldingWord-dev/door43.org/issues
+
+1. Gogs (Git website)
+
+When a repository is updated on `Gogs <https://test.door43.org:3000>`_, the commit triggers all webhooks in the repo's settings to be triggered. One of those webhooks, which our copy of Gogs sets up automatically for every new repo, is a call to https://test-api.door43.org/client/webhook (API Gateway -> Lambda function).
+
+2. Webhook (Lambda function - API Gateway triggered)
+
+The webhook triggered in Gogs (#1) sends the commit payload to the AWS API Gateway `client` stage and the `webhook` method which triggers the `client_webhook Lambda function <https://github.com/unfoldingWord-dev/tx-manager-lambda/blob/develop/functions/client_webhook/main.py>`_.
+
+The webhook function expects the following variables in the payload:
+
+* data - the commit payload from Gogs
+* api_url* - the base URL to the tX Manager API (e.g. https://test-api.door43.org)
+* pre_convert_bucket* - the S3 bucket in which to put the zip file of the files to be converted (e.g. tx-webhook)
+* cdn_bucket* - the S3 bucket in which the zip file of the converted files is to be found in client_callback (e.g. cdn.door43.org)
+* gogs_url* - the URL to the Gogs site to verify user token (e.g. https://test.door43.org:3000)
+* gogs_user_token* - a user token of a valid user to prove they are a user so we can track job requests (for the webclient we just have one user token for all requests, given by the API Gateway)
+
+*these variables are set up in the `'client' Stage Variables <https://us-west-2.console.aws.amazon.com/apigateway/home?region=us-west-2#/apis/94c6v76xoh/stages/client>`_, so dev and prod gateways can have different variables
+
+The client_webhook function is responsible for standardizing both a manifest.json file and the resource containers from all types of repos committed to Gogs. Converters (#4) expect the files to be converted to be in a flat-level zip file, where all files to be converted (with the input file extension) are one file per chapter (Bible, OBS) and in alphabetical order for logging and display purposes. Once the files are zipped up and the zip file put at https://test-cdn.door43.org/temp/<repo>/<commit>, the client webhook function requests a job by posting a request to https://test-api.door43.org/tx/job and exits.
+
+3. Request Job (Lamdbda function - API Gateway triggered)
+
+Request Job is triggered through a call to the AWS API Gateway which triggers the `request_job lambda function <https://github.com/unfoldingWord-dev/tx-manager-lambda/blob/develop/functions/tx-manager_request_job/main.py>`_. This function expects the following variables in the payload:
+
+* gogs_url* - the URL to the Gogs site to verify user token (e.g. https://git.door43.org)
+* api_url* - the base URL to the tX Manager API (e.g. https://test-api.door43.org)
+* data - information about the job to performed. It contains the following variables:
+
+   * gogs_user_token - a user token of a valid user to prove they are a user so we can traob requests
```

Comment:
I think I lost a block of text. The next line should be api_url, which vanished. Will add.